flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -484,7 +484,10 @@ protected Properties getConsumerConfiguration() {
 	 */
 	protected final void emitRecordAndUpdateState(T record, long recordTimestamp, int shardStateIndex, SequenceNumber lastSequenceNumber) {
 		synchronized (checkpointLock) {
-			sourceContext.collectWithTimestamp(record, recordTimestamp);
+			if (record != null) {
+				sourceContext.collectWithTimestamp(record, recordTimestamp);
```

Comment:
Are we silently skipping the record or do we log _somewhere_ that a record was invalid?

Reply:
We currently do not have a log for that.
I'll add a warning log if the record is null.